osu.Game/Online/SignalRUnionWorkaroundResolver.cs

```diff
@@ -20,34 +19,22 @@ public class SignalRUnionWorkaroundResolver : IFormatterResolver
         public static readonly MessagePackSerializerOptions OPTIONS =
             MessagePackSerializerOptions.Standard.WithResolver(new SignalRUnionWorkaroundResolver());
 
-        public static readonly IReadOnlyList<Type> BASE_TYPES = new[]
-        {
-            typeof(MatchServerEvent),
-            typeof(MatchUserRequest),
-            typeof(MatchRoomState),
-            typeof(MatchUserState),
-        };
-
-        public static readonly IReadOnlyList<Type> DERIVED_TYPES = new[]
-        {
-            typeof(ChangeTeamRequest),
-            typeof(TeamVersusRoomState),
-            typeof(TeamVersusUserState),
-        };
+        private static readonly IReadOnlyDictionary<Type, IMessagePackFormatter> formatter_map = createFormatterMap();
 
-        private static readonly IReadOnlyDictionary<Type, IMessagePackFormatter> formatter_map = new Dictionary<Type, IMessagePackFormatter>
+        private static IReadOnlyDictionary<Type, IMessagePackFormatter> createFormatterMap()
         {
-            { typeof(TeamVersusUserState), new TypeRedirectingFormatter<TeamVersusUserState, MatchUserState>() },
-            { typeof(TeamVersusRoomState), new TypeRedirectingFormatter<TeamVersusRoomState, MatchRoomState>() },
-            { typeof(ChangeTeamRequest), new TypeRedirectingFormatter<ChangeTeamRequest, MatchUserRequest>() },
+            IEnumerable<(Type derivedType, Type baseType)> baseDerived = SignalRWorkaroundTypes.BASE_DERIVED;
 
-            // These should not be required. The fallback should work. But something is weird with the way caching is done.
+            // This should not be required. The fallback should work. But something is weird with the way caching is done.
             // For future adventurers, I would not advise looking into this further. It's likely not worth the effort.
-            { typeof(MatchUserState), new TypeRedirectingFormatter<MatchUserState, MatchUserState>() },
-            { typeof(MatchRoomState), new TypeRedirectingFormatter<MatchRoomState, MatchRoomState>() },
-            { typeof(MatchUserRequest), new TypeRedirectingFormatter<MatchUserRequest, MatchUserRequest>() },
-            { typeof(MatchServerEvent), new TypeRedirectingFormatter<MatchServerEvent, MatchServerEvent>() },
-        };
+            baseDerived = baseDerived.Concat(baseDerived.Select(t => (t.baseType, t.baseType))).Distinct();
+
+            return new Dictionary<Type, IMessagePackFormatter>(baseDerived.Select(t =>
+            {
+                var formatter = (IMessagePackFormatter)Activator.CreateInstance(typeof(TypeRedirectingFormatter<,>).MakeGenericType(t.derivedType, t.baseType));
```

Comment:
I really don't feel comfortable with use of `Activator` here..

Reply:
`Activator.CreateInstance` should work on iOS as far as I'm aware, unless there's something more to it here?